test/test_in_toto_record.py

```diff
@@ -0,0 +1,185 @@
+    files_to_remove = [
+      "test-key", "test-key.pub", "test-step.link", self.test_artifact
+    ]
+    _try_remove_files(files_to_remove)
+
+
+  def test_command_start_stop_required_args(self):
+    """Test CLI command record start/stop with required arguments. """
+    with patch.object(sys, 'argv',  self.args + ["start"]):
+      in_toto_record_main()
+
+    with patch.object(sys, 'argv', self.args + ["stop"]):
+      in_toto_record_main()
+
+  def test_command_start_stop_optional_args(self):
+    """Test CLI command record start/stop with optional arguments. """
+    with patch.object(sys, 'argv', self.args + ["start", "--materials",
+        self.test_artifact]):
+      in_toto_record_main()
+
+    with patch.object(sys, 'argv', self.args + ["stop", "--products",
+        self.test_artifact]):
+      in_toto_record_main()
+
+
+
+
+
+class TestInTotoRecordStart(unittest.TestCase):
+  """"Test in_toto_record_start(step_name, key_path, material_list). """
+
+  @classmethod
+  def setUpClass(self):
+    """Generate key pair, dummy artifact and base arguments. """
+    self.step_name = "test-step"
+    self.key_path = "test-key"
+    generate_and_write_rsa_keypair(self.key_path)
+    self.test_material= "test-material"
+    open(self.test_material, "a").close()
+
+  @classmethod
+  def tearDownClass(self):
+    """Remove key pair, dummy artifact and link metadata. """
+    files_to_remove = [
+        self.key_path, self.key_path + ".pub", self.test_material,
+        "." + self.step_name + ".link-unfinished"
+      ]
+    _try_remove_files(files_to_remove)
+
+  def test_create_link_unfinished_file(self):
+    """Test record start creates a link metadata file. """
+    in_toto_record_start(self.step_name, self.key_path, [self.test_material])
+    open("." + self.step_name + ".link-unfinished", "r")
+
+  def test_record_test_material(self):
+    """Test record start records expected material. """
+    link = in_toto_record_start(
+        self.step_name, self.key_path, [self.test_material])
+    self.assertEquals(link.materials.keys(), [self.test_material])
+
+
+
+
+
+class TestInTotoRecordStop(unittest.TestCase):
+  """"Test in_toto_record_start(step_name, key_path, material_list). """
+
+  @classmethod
+  def setUpClass(self):
+    """Generate key pair, dummy artifact and base arguments. """
+    self.step_name = "test-step"
+    self.key_path = "test-key"
+    self.key_path2 = "test-key2"
+    generate_and_write_rsa_keypair(self.key_path)
+    generate_and_write_rsa_keypair(self.key_path)
+
+    self.test_product = "test-product"
+    open(self.test_product, "a").close()
+
+  @classmethod
+  def tearDownClass(self):
+    """Remove key pair, dummy artifact and link metadata. """
+    files_to_remove = [
+      self.key_path, self.key_path + ".pub", self.key_path2,
+      self.key_path2 + ".pub", self.test_product, self.step_name + ".link",
+      "." + self.step_name + ".link-unfinished"
```

Comment:
Why? This is only a list of file names. Anyway, I will change this to use a tmp directory as you suggested earlier.